Library build's watch mode breaks on branch-switch

This is because of the fact that on branch switch, the `gulpfile` is replaced with older gulpfile and the `watch` routine isn't the same anymore.

This is easily fixed by fetching the latest master's changes into the working branches (so that the gulpfiles are same across branches). After which this problem won't happen anymore.

To solve this programmatically, we'll have to add a git hook that overwrites Gulpfiles on branch changes, as temporary solution, until everyone has checked out the new code to their working branch. After which, this will again have to be removed.

This happens on a new branch which is up to date with master (has the new gulp file). Build mode breaks when I make code changes.

/home/nimisha/Desktop/GIT/almighty-ui/gulpfile.js:123
      mach.transpileTS();
           ^
TypeError: mach.transpileTS is not a function
    at FSWatcher.<anonymous> (/home/nimisha/Desktop/GIT/almighty-ui/gulpfile.js:123:12)
    at emitTwo (events.js:126:13)
    at FSWatcher.emit (events.js:214:7)
    at FSWatcher.<anonymous> (/home/nimisha/Desktop/GIT/almighty-ui/node_modules/gulp/node_modules/chokidar/index.js:198:15)
    at FSWatcher._emit (/home/nimisha/Desktop/GIT/almighty-ui/node_modules/gulp/node_modules/chokidar/index.js:240:5)
    at FSWatcher.<anonymous> (/home/nimisha/Desktop/GIT/almighty-ui/node_modules/gulp/node_modules/chokidar/lib/nodefs-handler.js:263:16)
    at FSReqWrap.oncomplete (fs.js:153:5)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! fabric8-planner@0.0.0-development build: `gulp build "--watch"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fabric8-planner@0.0.0-development build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-02-15T09_13_01_498Z-debug.log

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -43,6 +43,24 @@ var distPath  = 'dist';
 // Global namespace to contain the reusable utility routines
 let mach = {};
 
+// Serialized typescript compile and post-compile steps
+mach.transpileTS = function () {
+  return (gulp.series(function () {
+    return ngc('tsconfig.json');
+  }, function () {
+    // FIXME: why do we need that?
+    // Replace templateURL/styleURL with require statements in js.
+    return gulp.src(['dist/app/**/*.js'])
+    .pipe(replace(/templateUrl:\s/g, "template: require("))
+    .pipe(replace(/\.html',/g, ".html'),"))
+    .pipe(replace(/styleUrls: \[/g, "styles: [require("))
+    .pipe(replace(/\.less']/g, ".css').toString()]"))
+    .pipe(gulp.dest(function (file) {
+      return file.base; // because of Angular 2's encapsulation, it's natural to save the css where the less-file was
+    }));
+  }))();
+}
+
 // Copy files to the distPath
 mach.copyToDist = function (srcArr) {
   return gulp.src(srcArr)
@@ -85,22 +103,7 @@ gulp.task('build', function (done) {
 
   // app (default)
 
-  // Transpile *.ts to *.js; _then_ post-process require statements to load templates
-  (gulp.series(function () {
-    return ngc('tsconfig.json');
-  }, function () {
-    // FIXME: why do we need that?
-    // Replace templateURL/styleURL with require statements in js.
-    return gulp.src(['dist/app/**/*.js'])
-    .pipe(replace(/templateUrl:\s/g, "template: require("))
-    .pipe(replace(/\.html',/g, ".html'),"))
-    .pipe(replace(/styleUrls: \[/g, "styles: [require("))
-    .pipe(replace(/\.less']/g, ".css').toString()]"))
-    .pipe(gulp.dest(function (file) {
-      return file.base; // because of Angular 2's encapsulation, it's natural to save the css where the less-file was
-    }));
-  }))();
-
+  mach.transpileTS(); // Transpile *.ts to *.js; _then_ post-process require statements to load templates
   mach.transpileLESS(appSrc + '/**/*.less'); // Transpile and minify less, storing results in distPath.
   mach.copyToDist(['src/**/*.html']); // Copy template html files to distPath
   gulp.src(['LICENSE', 'README.adoc', 'package.json']).pipe(gulp.dest(distPath)); // Copy static assets to distPath

